contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -1749,8 +1751,26 @@ pltsql_sequence_datatype_map(ParseState *pstate,
 		return;
 
 	type_def = defGetTypeName(as_type);
+	type_names = type_def->names;
+	switch (list_length(type_names))
+	{
+		case 2:
+			strVal(linitial(type_names)) = get_physical_schema_name(get_cur_db_name(),strVal(linitial(type_names)));
```

Comment:
is it OK to change type names inside type_def->names

Reply:
The modified value is not being used in any of the parent functions ( i.e. init_params, DefineSequence, and ProcessUtilitySlow), therefore I believe it would not create any issues elsewhere.

Reply:
Surendra has a point, I think we should not change the state of "type_def" since it is extracted from "as_type" argument. We should remain transparent with the caller. Any change in the caller function, in future, could have an impact due to this change. We can rather copy this list locally and use it for our purpose inside pltsql_sequence_datatype_map.